pkg/operator/ceph/controller/spec.go

```diff
@@ -601,6 +603,40 @@ fi`,
 	}
 }
 
+// GenerateLivenessProbeExecDaemon generates a liveness probe that makes sure a daemon has a socket,
+// that it can be called, and that it returns 0
+func GenerateReadinessProbeExecDaemon(daemonType, daemonID string) *v1.Probe {
+	confDaemon := getDaemonConfig(daemonType, daemonID)
+	return &v1.Probe{
+		ProbeHandler: v1.ProbeHandler{
+			Exec: &v1.ExecAction{
+				// Run with env -i to clean env variables in the exec context
+				// This avoids conflict with the CEPH_ARGS env
+				//
+				// Example:
+				// env -i sh -c "ceph --admin-daemon /run/ceph/ceph-mgr.a.asok mgr_status"
+				//
+				Command: []string{
+					"env",
+					"-i",
+					"sh",
+					"-c",
+					fmt.Sprintf(`outp="$(ceph --admin-daemon %s %s 2>&1)"
+rc=$?
+if [ $rc -ne 0 ]; then
+  echo "ceph mgr daemon readiness check failed with the following output:"
+  echo "$outp" | sed -e 's/^/> /g'
```

Comment:
The output when the probe fails is format when the probe fails is not right, we need to provide some useful information:
In this moment , if we query the events of the pod we see something like:
```

│ Events:                                                                                                                                                                                                                                     │
│   Type     Reason          Age                From               Message                                                                                                                                                                    │
│   ----     ------          ----               ----               -------                                                                                                                                                                    │
│   Normal   Scheduled       105s               default-scheduler  Successfully assigned rook-ceph/rook-ceph-mgr-a-6b96758f67-nmwcr to mykube1-ctlplane-0.karmalabs.corp                                                                      │
│   Normal   AddedInterface  105s               multus             Add eth0 [10.244.0.214/24] from cbr0                                                                                                                                       │
│   Normal   Pulled          104s               kubelet            Container image "quay.io/ceph/ceph:v17.2.5" already present on machine                                                                                                     │
│   Normal   Created         104s               kubelet            Created container chown-container-data-dir                                                                                                                                 │
│   Normal   Started         104s               kubelet            Started container chown-container-data-dir                                                                                                                                 │
│   Normal   Pulled          103s               kubelet            Container image "quay.io/ceph/ceph:v17.2.5" already present on machine                                                                                                     │
│   Normal   Created         103s               kubelet            Created container mgr                                                                                                                                                      │
│   Normal   Started         103s               kubelet            Started container mgr                                                                                                                                                      │
│   Normal   Pulled          103s               kubelet            Container image "jolmomar/rook:latest" already present on machine                                                                                                          │
│   Normal   Created         103s               kubelet            Created container watch-active                                                                                                                                             │
│   Normal   Started         103s               kubelet            Started container watch-active                                                                                                                                             │
│   Normal   Pulled          103s               kubelet            Container image "quay.io/ceph/ceph:v17.2.5" already present on machine                                                                                                     │
│   Normal   Created         103s               kubelet            Created container log-collector                                                                                                                                            │
│   Normal   Started         103s               kubelet            Started container log-collector                                                                                                                                            │
│   Warning  Unhealthy       5s (x12 over 85s)  kubelet            Readiness probe failed: ceph mgr daemon readiness check failed with the following output:                                                                                  │
│ > no valid command found; 10 closest matches:                                                                                                                                                                                               │
│ > 0                                                                                                                                                                                                                                         │
│ > 1                                                                                                                                                                                                                                         │
│ > 2                                                                                                                                                                                                                                         │
│ > abort                                                                                                                                                                                                                                     │
│ > assert                                                                                                                                                                                                                                    │
│ > config diff                                                                                                                                                                                                                               │
│ > config diff get <var>                                                                                                                                                                                                                     │
│ > config get <var>                                                                                                                                                                                                                          │
│ > config help [<var>]                                                                                                                                                                                                                       │
│ > config set <var> <val>...                                                                                                                                                                                                                 │
│ > admin_socket: invalid command     
```

I suggest to remove the output generated by the command in the standby manager and use a more clear message:

```
if [ $rc -ne 0 ]; then
  echo "Standby manager does not process requests"
  exit $rc
fi`,
```
In this way we will have this when the readiness probe fails:

```
│ Events:                                                                                                                                                                                                                                     │
│   Type     Reason          Age                From               Message  
|   Warning  Unhealthy       5s (x12 over 85s)  kubelet            Readiness probe failed: Standby manager does not process requests
```

Reply:
Good idea so we give a clear message to the user about why readiness is failing. I changed the message.